File: exercise/ex02/e1_fibo/solution/Fibo.java
import java.io.*;
import java.util.*;

 public   class   Fibo   {
        public static int fibo(int n) {
                if (n < 2) return n;
                int a = 0;
                int b = 1;
                for (int i = 2;i <= n;i++) {
                   int c = a + b;
                   a = b;
                   b = c;
                }
                return b;
        }

        public static void main(String[] argv) {
                Scanner sc = new Scanner(System.in);
                int n = sc.nextInt();
                System.out.println(fibo(n));
        }
}

File: class/dp01/fibo.cpp
#include <iostream>
#include <vector>

using namespace std;

long long int fibo(long long int n) {
  if (n <= 2) return 1;
  int a = fibo(n-1);
  int b = fibo(n-2);
  return a+b;
}

int main() {
  long long int n =50;
  cout << fibo(n);
}

